### ISSUE TITLE
problem of running resnet-50

### ISSUE THREAD

**Issue by fastalgo** [intel/caffe#119]
remove_batch_norm.cpp:276] Dropped Layer: layer_512_3_bn3

**Comment by jgong5** [comment]
May I know what's the problem you referring to?

**Comment by fastalgo** [comment]
why drop the batch norm layer?

**Comment by jgong5** [comment]
It is an optimization. In the test phase, batch norm layer is fused with convolution. This would speed up the inference of the models with batch norms.

**Comment by fastalgo** [comment]
How can I keep this layer?

**Comment by fastalgo** [comment]
I mean, in the training part.

**Comment by jgong5** [comment]
BatchNorm is removed in training phase.

**Comment by jgong5** [comment]
BatchNorm is not removed in training phase.

**Comment by fastalgo** [comment]
Okay, thanks!

**Comment by fastalgo** [comment]
Hi, how can I keep the batch norm layer in the test phase?

**Comment by jgong5** [comment]
We don't provide a way right now. May I know why you want to keep batch norm?

**Comment by fastalgo** [comment]
Looks like the training loss is low but the test loss is high.

**Comment by jgong5** [comment]
I don't think that a problem of batch norm removal. It should be mathematically equivalent. What model are you training?

**Comment by fastalgo** [comment]
Okay, thanks, let me try again. If it doesn't work, I will come back. I just wrote my own model. :-)

**Comment by etaf** [comment]
Please reopen this issue if you get problems again.

**Comment by fastalgo** [comment]
Thanks!
